jqjq.jq

```diff
@@ -2543,6 +2545,187 @@ def eval($expr; $globals; $builtins_env):
+    // option(null; "tab"; .indent = "tab" | .print_pretty = true)
+    // option(null; "indent"; handle_indent)
+    # // option(null; "seq"; .seq = true)
+    # // option(null; "stream"; .parse_streaming = true)
+    # // option(null; "stream-errors"; (.parse_streaming, .parse_stream_errors) = true)
+    # // option("e"; "exit-status"; .exit_status = true)
+    // option(null; "args"; .args.rest_are_positional = "arg")
+    // option(null; "jsonargs"; .args.rest_are_positional = "argjson")
+    // option(null; "arg"; handle_arg("arg"; "value"))
+    // option(null; "argjson"; handle_arg("argjson"; "text"))
+    // option(null; "rawfile"; handle_arg("rawfile"; "filename"))
+    // option(null; "slurpfile"; handle_arg("slurpfile"; "filename"))
+    # // option(null; "debug-dump-disasm"; .debug_dump_disasm = true)
+    # // option(null; "debug-trace=all"; .debug_trace_all = true)
+    # // option(null; "debug-trace"; .debug_trace = true)
+    // option("h"; "help"; .action = "help")
+    # // option("V"; "version"; .action = "version")
+    # // option(null; "build-configuration"; .action = "build-configuration")
+    // option(null; "run-tests"; .action = "run-tests")
+    # jqjq extensions:
+    // option(null; "jq"; handle_jq)
+    // option(null; "repl"; .mode = "repl")
+    // option(null; "lex"; .mode = "lex")
+    // option(null; "parse"; .mode = "parse")
+    // option(null; "no-builtins"; .no_builtins = true)
+    //
+      ( if .args.is_short then "-\(.args.curr[:1])" else "--\(.args.curr)" end
+      | "Unknown option: \(.)" | die
+      )
+    );
+  def parse_options_in_arg:
+    ( parse_option
+    | if .args.curr != null and .action == null then parse_options_in_arg end
+    );
+  def parse_arg:
+    ( .args.rest[0] as $arg
+    | .args.rest = .args.rest[1:]
+    | if .args.done or ($arg | test("^-[\\-a-zA-Z]") | not) then
+        if .program == null then
+          .program = $arg
+        elif .args.rest_are_positional != null then
+          .program_args.positional += [{type: .args.rest_are_positional, value: $arg}]
+        else
+          .files += [$arg]
+        end
+      elif $arg == "--" then
+        .args.done = true
+      else
+        ( if $arg | startswith("--") then
+            .args.curr = $arg[2:] | .args.is_short = false
+          else
+            .args.curr = $arg[1:] | .args.is_short = true
+          end
+        | parse_options_in_arg
+        )
+      end
+    );
+  def parse_args:
+    if (.args.rest | length > 0) and .action == null then
+      parse_arg | parse_args
+    else del(.args)
+    end;
+  ( {
+      args: {
+        rest: .,
+      },
+      program_args: {
+        positional: [],
+        named: {},
+      },
+      indent: 2,
+      print_pretty: true,
+    }
+  | parse_args
+  );
+
+def invoke_client_jqjq:
+  def sh_escape:
+    if . == "" or test("[' \n\\\\]") then
+      "'" + gsub("'"; "'\\''") + "'"
+    end;
+  ( $ARGS.positional
+  | parse_options
+  | [ (.jq // "jq" | sh_escape)
```

Comment:
I didn't know about that!

Reply:
Actually, instead of using that, I'm going to fix escaping of `$`. I don't like that `@sh` always quotes strings:
```c
line = jv_string_append_str(line, "'");
line = jv_string_concat(line, escape_string(x, "''\\''\0"));
line = jv_string_append_str(line, "'");
```